nextjs-app-router/app/layout.js

```diff
@@ -0,0 +1,52 @@
+// This file defines the overall layout of all pages in the application.
+// The default function exported from this module must return a document with
+// `html` and `body` tags. A `head` tag will be ignored.
+
+import Script from 'next/script'
+import Link from 'next/link'
+import newrelic from 'newrelic'
+
+// Somehow, this ends up including our static stylesheet correctly.
+// See https://nextjs.org/docs/app/building-your-application/styling/css-modules#global-styles.
+import './style.css'
+
+export default async function RootLayout({ children }) {
+  if (newrelic.agent.collector.isConnected() === false) {
+    await new Promise((resolve) => {
+      newrelic.agent.on("connected", resolve)
+    })
+  }
+
+  const browserTimingHeader = newrelic.getBrowserTimingHeader({
+    hasToRemoveScriptWrapper: true,
+    allowTransactionlessInjection: true,
+  })
+
+  return (
+    <html>
+    <body>
+    <ul className="navbar">
+      <li><a href="/">Home</a></li>
+      <li><Link href="/users" key={"users"}>Users</Link></li>
+      <li><Link href="/about" key={"about"}>About</Link></li>
+    </ul>
+    {children}
+
+    <Script
```

Comment:
One thing to note: we have no control over the order in which the script is added to the document, only that it gets added to the `head` block. The result, is that the browser agent script is loaded last in this example.

Reply:
good work figuring this out 👏🏻